pkg/assembler/backends/arangodb/certifyLegal_test.go

```diff
@@ -0,0 +1,783 @@
+				Attribution: ptrfrom.String("Copyright Jeff"),
+			},
+			ExpLegal: []*model.CertifyLegal{
+				{
+					Subject:     p1out,
+					Attribution: "Copyright Jeff",
+				},
+			},
+		},
+		{
+			Name:  "Query on time",
+			InPkg: []*model.PkgInputSpec{p1},
+			Calls: []call{
+				{
+					PkgSrc: model.PackageOrSourceInput{
+						Package: p1,
+					},
+					Legal: &model.CertifyLegalInputSpec{
+						TimeScanned: t3,
+					},
+				},
+				{
+					PkgSrc: model.PackageOrSourceInput{
+						Package: p1,
+					},
+					Legal: &model.CertifyLegalInputSpec{
+						TimeScanned: t2,
+					},
+				},
+			},
+			Query: &model.CertifyLegalSpec{
+				TimeScanned: &t2,
+			},
+			ExpLegal: []*model.CertifyLegal{
+				{
+					Subject:     p1out,
+					TimeScanned: t2,
+				},
+			},
+		},
+		{
+			Name:  "Query multiple",
+			InPkg: []*model.PkgInputSpec{p1, p2, p3},
+			InLic: []*model.LicenseInputSpec{l1},
+			Calls: []call{
+				{
+					PkgSrc: model.PackageOrSourceInput{
+						Package: p1,
+					},
+					Dec: []*model.LicenseInputSpec{l1},
+					Legal: &model.CertifyLegalInputSpec{
+						Justification: "test justification special",
+					},
+				},
+				{
+					PkgSrc: model.PackageOrSourceInput{
+						Package: p2,
+					},
+					Dec: []*model.LicenseInputSpec{l1},
+					Legal: &model.CertifyLegalInputSpec{
+						Justification: "test justification special",
+					},
+				},
+				{
+					PkgSrc: model.PackageOrSourceInput{
+						Package: p3,
+					},
+					Dec: []*model.LicenseInputSpec{l1},
+					Legal: &model.CertifyLegalInputSpec{
+						Justification: "test justification other",
+					},
+				},
+			},
+			Query: &model.CertifyLegalSpec{
+				Justification: ptrfrom.String("test justification special"),
+			},
+			ExpLegal: []*model.CertifyLegal{
+				{
+					Subject:          p1out,
+					DeclaredLicenses: []*model.License{l1out},
+					Justification:    "test justification special",
+				},
+				{
+					Subject:          p2out,
+					DeclaredLicenses: []*model.License{l1out},
+					Justification:    "test justification special",
+				},
+			},
+		},
+		// {
+		// 	Name: "Ingest without Package",
+		// 	Calls: []call{
+		// 		{
+		// 			PkgSrc: model.PackageOrSourceInput{
+		// 				Package: pNone,
+		// 			},
+		// 			Legal: &model.CertifyLegalInputSpec{},
+		// 		},
+		// 	},
+		// 	ExpIngestErr: true,
+		// },
+		// {
+		// 	Name: "Ingest without Source",
+		// 	Calls: []call{
+		// 		{
+		// 			PkgSrc: model.PackageOrSourceInput{
+		// 				Source: sNone,
+		// 			},
+		// 			Legal: &model.CertifyLegalInputSpec{},
+		// 		},
+		// 	},
+		// 	ExpIngestErr: true,
+		// },
+		// {
+		// 	Name:  "Ingest without License",
+		// 	InPkg: []*model.PkgInputSpec{p1},
+		// 	Calls: []call{
+		// 		{
+		// 			PkgSrc: model.PackageOrSourceInput{
+		// 				Package: p1,
+		// 			},
+		// 			Dec:   []*model.LicenseInputSpec{lNone},
+		// 			Legal: &model.CertifyLegalInputSpec{},
+		// 		},
+		// 	},
+		// 	ExpIngestErr: true,
```

Comment:
@pxp928 These missing-noun expected errors are hanging the test. Am I doing something wrong in the query?

Reply:
yes we can remove these as they got moved to the resolvers

Reply:
yeah, the test will timeout (exponential backoff on the query) if the subject is not already ingested. However, the assembler will enforce the ingestion of the nouns before the verbs. Not sure if we need to also test it here.

Reply:
Sounds good, opened #1360 to track going forward.